python/ctsm/subset_data.py

```diff
@@ -0,0 +1,893 @@
+at either a single point or a region using the global dataset.
+
+After creating a case using a global compset, run preview_namelist.  
+From the resulting lnd_in file in the run directory, find the name 
+of the domain file, and the surface data file.
+
+From the datm streams files (e.g. datm.streams.txt.CLMGSWP3v1.Precip)
+find the name of the datm forcing data domain file and forcing files.  
+Use these file names as the sources for the single point/regional
+files to  be created (see below).
+
+After running this script, point to the new CLM domain and surface 
+dataset using the user_nl_clm file in the case directory.  In addition, 
+copy the datm.streams files to the case directory, with the prefix 
+'user_', e.g. user_datm.streams.txt.CLMGSWP3v1.Precip.  Change the 
+information in the user_datm.streams* files to point to the single 
+point datm data (domain and forcing files) created using this script.  
+
+The domain file is not set via user_nl_clm, but requires changing 
+LND_DOMAIN and ATM_DOMAIN (and their paths) in env_run.xml.  
+
+Using single point forcing data requires specifying the nearest 
+neighbor mapping algorithm for the datm streams (usually they are 
+the first three in the list) in user_nl_datm: mapalgo = 'nn','nn','nn', 
+..., where the '...' can still be 'bilinear', etc, depending on the 
+other streams that are being used, e.g. aerosols, anomaly forcing, 
+bias correction.
+
+The file env_mach_pes.xml should be modified to specify a single 
+processor.  The mpi-serial libraries should also be used, and can be 
+set in env_build.xml by changing "MPILIB" to "mpi-serial" prior to 
+setting up the case.  
+
+The case for the single point simulation should have river routing 
+and land ice models turned off (i.e. the compset should use stub 
+models SROF and SGLC)
+
+By default, it only extracts surface dataset and for extracting other
+files, the appropriate flags should be used.
+-------------------------------------------------------------------
+To run the script for a single point:
+    ./subset_data.py point --help
+ 
+To run the script for a region:
+    ./subset_data.py reg --help 
+
+To remove NPL from your environment on Cheyenne/Casper:
+    deactivate
+-------------------------------------------------------------------
+"""
+
+# TODO
+# Automatic downloading of missing files if they are missing
+# default 78 pft vs 16 pft
+
+#  Import libraries
+from __future__ import print_function
+
+import sys
+import os
+import string
+import logging
+import subprocess
+import argparse
+
+import numpy as np
+import xarray as xr
+
+from datetime import date
+from getpass import getuser
+from logging.handlers import RotatingFileHandler
+from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
+
+# -- add python/ctsm  to path
+_CTSM_PYTHON = os.path.join(
+    os.path.dirname(os.path.realpath(__file__)), os.pardir, os.pardir, "python"
+)
+
+sys.path.insert(1, _CTSM_PYTHON)
```

Comment:
- [ ] I would get rid of this path manipulation here and just have it in the little wrapper script. Having it in both places seems unnecessary and I think will lead to the python directory being added twice in sys path (not necessarily a problem, but unnecessary). My convention has been that the things in python/ctsm cannot be run as executables: they need to be run from one of the top-level wrapper scripts. (So, along with this, I'm recommending getting rid of the name == main check at the bottom.)